### ISSUE TITLE
Integrate `sleap-nn`

### ISSUE THREAD

**Issue by gitttt-1234** [talmolab/sleap#2251]
This issue outlines the migration from TensorFlow backend to SLEAP-NN (PyTorch-based backend) to address dependency conflicts and Windows+CUDA support issues. The migration involves removing the entire sleap/nn folder and integrating sleap-nn as a pip dependency.

- [x] Add config dialog windows classes #2249 
- [x] Add `sleap-nn` CLIs in `runners.py` #2249 
- [ ] Saving predictions to the slp file doesnt work after inference (issue with merging) #2255 
- [ ] Use Metrics from `sleap-nn` to display the Metric Evaluation for Trained Models #2254 
- [ ] Test if all field in training window works
- [ ] Test if all fields in inference works
- [ ] Make sure the GUI can be pre-filled by loading an existing `training_config.json` or `training_config.yaml`
        - Right now, this is not working, because of `cattr` (maybe a version issue)
- [ ] Ensure Resume training/ loading pretrained weights/ loading training config works from GUI
- [x] Remove config parameters from GUI depending on `sleap-nn` config
- [ ] Add config parameters to GUI depending from `sleap-nn` config
              - `data_config`: data_pipeline_fw, augmentation
              - `model_config`: convnext/ swint backbone architectures
              - `trainer_config`: num_workers, trainer_devices, trainer_accelerator, wandb_config
              - #2281

**Comment by gitttt-1234** [comment]
continued in #2297
